### PR TITLE
Add support for button dropdown component

### PR DESCRIPTION
**Pull request by phndiaye**
### What does this PR do?

Add support for button dropdown component

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [ ] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/button-dropdown.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  @action
+  toggleDropdown(event: MouseEvent) {
+    event.stopPropagation();
+    this.displayDropdown = !this.displayDropdown;
+  }
+
+  @action
+  closeDropdown() {
```
Comment (Miexil) [inline addon/components/o-s-s/button-dropdown.ts:27]: `: void`

File: app/styles/molecules/button-dropdown.less [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    i,
+    span {
+      color: var(--color-gray-900);
+    }
+  }
+
+  .oss-button-dropdown__items {
+    display: none;
```
Comment (Miexil) [inline app/styles/molecules/button-dropdown.less:27]: Handling this with `display: none` & `display: flex` in CSS makes the options always visible in the dom-elements. Is that wanted ?
Why not go with the `if` in the handlebars from ember ?

File: tests/integration/components/o-s-s/button-dropdown-test.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { click, render, setupOnerror } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+
+module('Integration | Component | o-s-s/button-dropdown', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it throws an error if nor icon or label args are provided', async function (assert) {
```
Comment (Miexil) [inline tests/integration/components/o-s-s/button-dropdown-test.ts:9]: `if no icon or label`

File: addon/components/o-s-s/button-dropdown.hbs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    </div>
+
+    <div class="fx-row fx-xalign-center" role="button" {{on "click" this.toggleDropdown}}>
+      <i class="far fa-caret-down"></i>
+    </div>
+  </div>
+
+  <div class="oss-button-dropdown__items {{if this.displayDropdown 'oss-button-dropdown__items--visible'}}"
+        {{on-click-outside this.closeDropdown}}>
+    {{yield to="items"}}
```
Comment (Miexil) [inline addon/components/o-s-s/button-dropdown.hbs:20]: When you click on a yielded item, does it close the dropdown menu ?

File: addon/components/o-s-s/button-dropdown.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  @tracked displayDropdown: boolean = false;
+
+  constructor(owner: unknown, args: OSSButtonDropdownArgs) {
+    super(owner, args);
+
+    assert('[component][OSS::ButtonDropdown] You must pass either a @label or an @icon argument.', args.label || args.icon);
+  }
+
+  @action
+  toggleDropdown(event: MouseEvent) {
```
Comment (Miexil) [inline addon/components/o-s-s/button-dropdown.ts:21]: `: void`

File: addon/components/o-s-s/button-dropdown.hbs (hunk 4 above)
Comment (JulienVannier66) [reply addon/components/o-s-s/button-dropdown.hbs:20]: I think not

File: addon/components/o-s-s/button-dropdown.ts (hunk 5 above)
Comment (JulienVannier66) [inline addon/components/o-s-s/button-dropdown.ts:21]:
```suggestion
  toggleDropdown(event?: MouseEvent) {
```
And use `event?.stopPropagation();` what do you think?

File: app/styles/molecules/button-dropdown.less [hunk 6]
```diff
@@ -0,0 +1,62 @@
+.oss-button-dropdown {
+  position: relative;
+
+  &__trigger {
+    border: 1px solid @upf-primary-rock-blue;
```
Comment (JulienVannier66) [inline app/styles/molecules/button-dropdown.less:5]: Maybe use new color pallet

File: app/styles/molecules/button-dropdown.less [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    height: 37px;
+    padding: var(--spacing-px-6) 0;
+
+    div {
+      height: 37px;
+      padding: 0 var(--spacing-px-18);
+
+      &:last-child {
+        padding: 0 var(--spacing-px-12);
+        border-left: 1px solid @upf-primary-rock-blue;
```
Comment (JulienVannier66) [inline app/styles/molecules/button-dropdown.less:16]: Same here for the color pallet

File: app/styles/molecules/button-dropdown.less [hunk 8]
```diff
@@ -0,0 +1,62 @@
+.oss-button-dropdown {
+  position: relative;
+
+  &__trigger {
+    border: 1px solid @upf-primary-rock-blue;
+    border-radius: var(--border-radius-sm);
+    height: 37px;
```
[the remark this answers is not in the document]
Comment (phndiaye) [reply app/styles/molecules/button-dropdown.less:7]: yep!

File: addon/components/o-s-s/button-dropdown.hbs (hunk 4 above)
Comment (phndiaye) [reply addon/components/o-s-s/button-dropdown.hbs:20]: nope! :/ a bit hard to trigger considering they are yielded outside the context of the component :/

File: app/styles/molecules/button-dropdown.less (hunk 6 above)
Comment (phndiaye) [reply app/styles/molecules/button-dropdown.less:5]: it doesn't much the existing buttons so it looks weird when placed next to existing buttons (checked w/ Victor)

File: app/styles/molecules/button-dropdown.less (hunk 7 above)
Comment (phndiaye) [reply app/styles/molecules/button-dropdown.less:16]: it doesn't much the existing buttons so it looks weird when placed next to existing buttons (checked w/ Victor)

File: addon/components/o-s-s/button-dropdown.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+interface OSSButtonDropdownArgs {
+  icon?: string;
+  label?: string;
+}
+
+export default class extends Component<OSSButtonDropdownArgs> {
+  @tracked displayDropdown: boolean = false;
+
+  menuContainer?: HTMLElement;
```
Comment (JulienVannier66) [inline addon/components/o-s-s/button-dropdown.ts:14]: You don't use this variable

**Review by JulienVannier66 — APPROVED** [review]
Otherwise LGTM

**Pull request merged**